### PR TITLE
make filesystem inodes read-only

### PR DESCRIPTION
**Pull request by Mic92**
Some tools can complain about them beeing writable. Since processes cannot write them anyway, that seems like a good default.

Untested. See https://github.com/Mic92/envfs/issues/166

### CODE REVIEW

File: src/fs.rs [hunk 1]
```diff
@@ -250,7 +250,7 @@ fn symlink_attr(ino: u64) -> FileAttr {
         crtime: UNIX_EPOCH,
         uid: 0,
         gid: 0,
-        perm: 777,
```
Comment (con-f-use) [inline src/fs.rs:253]: You sure it doesn't need to be `555`?

File: src/fs.rs [hunk 2]
```diff
@@ -250,7 +250,7 @@ fn symlink_attr(ino: u64) -> FileAttr {
...
+        perm: 444,
```
[the remark this answers is not in the document]
Comment (Mic92) [reply src/fs.rs:253]: Mhm. Actually I have never thought about permission on symlinks when it comes to execve().
I wonder if it's even posix compliant to have some != 777.

File: src/fs.rs [hunk 3]
```diff
@@ -45,7 +45,7 @@ const ROOT_DIR_ATTR: FileAttr = FileAttr {
     ctime: UNIX_EPOCH,
     crtime: UNIX_EPOCH,
     kind: FileType::Directory,
-    perm: 0o755,
+    perm: 0o555,
```
Comment (Mic92) [inline src/fs.rs:48]: I think this is fine to have as a default.

File: src/fs.rs [hunk 4]
```diff
@@ -250,7 +250,7 @@ fn symlink_attr(ino: u64) -> FileAttr {
         crtime: UNIX_EPOCH,
         uid: 0,
         gid: 0,
-        perm: 777,
+        perm: 0o555,
```
Comment (Mic92) [inline src/fs.rs:253]: I have to see if this is something that I can do by default or if this should be rather hidden behind a flag.

**Comment by Mic92** [comment]
After thinking about it, since I don't know why problem exist in the first place, I would not merge it for now.

**Comment by Mic92** [comment]
I have not seen any symlink implementation that was not using 777 as permissions.

**Pull request closed without merging**